### DISCUSSION TITLE
Converting V Strings To C Strings

### DISCUSSION THREAD

**Discussion by Wajinn** [vlang/v#13092] [General]
The documentation shows functions to convert C strings to V strings.  What about the other way around? Converting V strings to C strings for use with C functions?  Is there some function or easier known method of doing that?

**Comment by JalonSolov** [comment]
V strings are structures with 2 fields, `str` and `len`.  The `str` field is a NUL-terminated array of chars, which _is_ a C string.  So to pass a V string to C, just pass `my_string.str`.

You can also declare C string contants in V with the `c` prefix, as in `c'This is a C string'`.

**Reply by Wajinn** [reply]
Thanks, that would work in simpler cases, but what about the matching of other more complex types?

For instance, LPCSTR or LPTSTR, does anybody know what would be the vlang equivalent for such?

**Reply by JalonSolov** [reply]
Depends on what the C code wants to do with them, as well as what you want to do with them in V.  V doesn't make any distinction between a `CSTR` and a `TSTR` - every string can have UNICODE in them (since ASCII is just a subset of UNICODE).
